argo/providers/base/provider_app.go

```diff
@@ -307,6 +313,41 @@ func (providerApp ProviderApp[Config]) Run(provider Provider[Config], customComm
 				},
 			},
 		},
+		{
+			Name: "eventsource-server",
+			Flags: []cli.Flag{cli.StringFlag{
+				Name:     ProviderConstants.EventsourceServerPort,
+				Required: true,
+			}},
+			Action: func(c *cli.Context) error {
+				logger := LoggerFromContext(providerApp.Context)
+				providerConfig, err := providerApp.LoadProviderConfig(c)
+
+				lis, err := net.Listen("tcp", fmt.Sprintf("0.0.0.0:%d", c.Int(ProviderConstants.EventsourceServerPort)))
+				if err != nil {
+					logger.Error(err, "failed to listen")
+					os.Exit(1)
```

Comment:
Isn't there a way in the logger to log and exit or have I dreamt about it? In case I haven't, here and below, of course.

Reply:
Not `logr.Logger` but the go Logger has `log.Panic`